ts2wasm/src/backend/index.ts

```diff
@@ -9,3 +9,95 @@ export abstract class Ts2wasmBackend {
     public abstract emitText(options?: any): string;
     public abstract dispose(): void;
 }
+
+
+export interface SegmentInfo {
```

Comment:
This file is used to describe the common base class for all backends, it shouldn't contain specific implementations.

The `DataSegmentContext` is only used in `binaryen` backend, it would be better to keep this inside `binaryen` folder